src/proc/mod.rs

```diff
@@ -308,6 +294,46 @@ impl super::TypeInner {
             _ => false,
         }
     }
+
+    pub const fn scalar_bool() -> Self {
```

Comment:
I remember using it somewhere but seems it's unused right now (even though it could be used in a few places). I'll take it out and we can add it in a separate PR.